app/services/foreman_discovery/host_converter.rb

```diff
@@ -19,6 +19,43 @@ def self.to_managed(original_host, set_managed = true, set_build = true, added_a
     host
   end
 
+  def self.eventually_make_bond(host)
```

Comment:
Can you move this method out of HostConverter into LldpNeighbors or somewhere else? I suggest creating a "post processing framework":

* ForemanDiscovery::ImportHookService - a simple registry of hooks
* ForemanDiscovery::ImportHook - base class for processor - keep it simple `after_populate(host, facts)` as the only method for now
* ForemanDiscovery::ImportHooks::LldpNeighbor - implementation - move this method there

Optionally break method `populate_discovery_fields_from_facts` into three other separate hooks: DiscoveryAttribute, SubnetAndTaxonomy and LockTemplates. I can do this later if you don't want to touch this.

The reason I want this is in the future we might break the processing of NICs and Facts in Core in similar way, so we could easily merge this. The idea is not to save anything into database during the whole import phase so plugins can hook into the process easily and at the very end of the whole processing, save all NICs and fields to database in a single place.